include/assert.h

```diff
@@ -152,6 +152,8 @@
 #  endif
 #endif
 
+#define COMPILE_TIME_ASSERT(x) static_assert(x, "compile time assert failed")
```

Comment:
Which third-party library is using this macro?

Reply:
COMPILE_TIME_ASSERT is not a standard implementation, right? I have seen some repositories have similar implementations but with different parameters:
https://github.com/RabbitBio/RabbitFX/blob/a9c64920063e14c9d39cda5306320ab4b69cf1f5/io/Common.h#L47

Reply:
> Which third-party library is using this macro?

This is used by optee framework, the origin COMPILE_TIME_ASSERT macro implementation cannot compile on nuttx. In order to use optee while minimizing modifications to the optee repo, so we provide our own implementaion of COMPILE_TIME_ASSERT in NuttX

Reply:
> COMPILE_TIME_ASSERT is not a standard implementation, right? I have seen some repositories have similar implementations but with different parameters: https://github.com/RabbitBio/RabbitFX/blob/a9c64920063e14c9d39cda5306320ab4b69cf1f5/io/Common.h#L47

yes, different platform may have different implementation, a better implementation should be static_assert

Reply:
@extinguish are you planning to add lib optee at NuttX Apps? I think it should be a great addition